docs/coil.md

```diff
@@ -0,0 +1,154 @@
+
+Image(
+    painter = rememberCoilPainter("https://picsum.photos/300/300"),
+    contentDescription = "My content description",
+)
+```
+
+This painter loads the data passed in, using [Coil][coil], and then draws the resulting image.
+
+You can also customize the Coil [`ImageRequest`](https://coil-kt.github.io/coil/image_requests/) through the `requestBuilder` parameter. This allows usage of things like (but not limited to) transformations:
+
+```kotlin
+import androidx.compose.foundation.Image
+import com.google.accompanist.coil.rememberCoilPainter
+
+Image(
+    painter = rememberCoilPainter(
+        request = "https://picsum.photos/300/300",
+        requestBuilder = {
+            transformations(CircleCropTransformation())
+        },
+    ),
+    contentDescription = "My content description",
+)
+```
+
+## Fade-in animation
+
+This library has built-in support for animating loaded images in, using a [fade-in animation](https://material.io/archive/guidelines/patterns/loading-images.html).
+
+![](crossfade.gif)
+
+
+A `fadeIn: Boolean` parameter is available on [`rememberCoilPainter()`][rememberpainter] (default: `false`). When enabled, a default fade-in animation will be used when the image is successfully loaded:
+
+``` kotlin
+import androidx.compose.foundation.Image
+import com.google.accompanist.coil.rememberCoilPainter
+
+Image(
+    painter = rememberCoilPainter(
+        "https://picsum.photos/300/300",
+        fadeIn = true
+    ),
+    contentDescription = "My content description",
+)
+```
+
+## Custom content
+
+Some times you may wish to display some alternative content whilst the image is loading, or an error has occurred. The painter returned from `rememberCoilPainter()` is an instance of [`LoadPainter`][loadpainter], which is stateful and allows you to display different content as required:
+
+
+``` kotlin
+val painter = rememberCoilPainter("https://picsum.photos/300/300")
+
+Box {
+    Image(
+        painter = painter,
+        contentDescription = "My content description",
+    )
+
+    if (painter.loadState == ImageLoadState.Loading) {
+        // Display a circular progress indicator whilst loading
+        CircularProgressIndicator(Modifier.align(Alignment.Center))     
+    }
+}
+```
+
+[`ImageLoadState`][imageloadstate] has a number of different states, so tweak your logic to suit. You could also use a [`Crossfade()`][crossfade] or any other custom animation.
+
+## Previews
+
+To support Android Studio [Composable Previews](https://developer.android.com/jetpack/compose/tooling), you can provide a drawable resource ID via the `previewPlaceholder` parameter. That drawable will then be displayed when your content is displayed as a preview:
+
+```kotlin
+Image(
+    painter = rememberCoilPainter(
+        request = "https://picsum.photos/300/300",
+        previewPlaceholder = R.drawable.placeholder,
+    ),
+    contentDescription = "My content description",
+)
+```
+
```

Comment:
Worth calling out that preview placeholders don't need to be in the main source set?

Reply:
Done